### PR TITLE
[ALLUXIO-1904] Changed creation behavior of RemoteBlockReader in RemoteBlockInStream

### PR DESCRIPTION
**Pull request by HorizonNet**
https://alluxio.atlassian.net/browse/ALLUXIO-1904

### CODE REVIEW

**Comment by calvinjia** [comment]
@HorizonNet Thanks for the updated PR, LGTM

File: core/client/src/main/java/alluxio/client/block/RemoteBlockInStream.java [hunk 1]
```diff
@@ -102,6 +104,10 @@ public void close() throws IOException {
       mClosed = true;
       mCloser.close();
     }
+
+    if (mReader != null) {
+      mReader.close();
```
Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/block/RemoteBlockInStream.java]: Can you put it in the mCloser?

**Comment by peisun1115** [comment]
@HorizonNet  Thanks for contributing. The cost of creating a new NettyRemoteBlockReader is very small. I am actually fine without making this change. The PR itself looks OK. One comment left.

**Comment by HorizonNet** [comment]
@peisun1115 Thanks for the feedback. I updated the PR accordingly. Hope I got your point right.

**Comment by peisun1115** [comment]
@HorizonNet  LGTM

**Pull request merged**